File: app/src/main/java/dev/voleum/ordermolder/SettingsActivity.java
package dev.voleum.ordermolder;

import android.os.Bundle;
import android.widget.EditText;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.appcompat.widget.Toolbar;
import androidx.preference.ListPreference;
import androidx.preference.Preference;

import com.takisoft.preferencex.EditTextPreference;
import com.takisoft.preferencex.PreferenceFragmentCompat;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settings, new SettingsFragment())
                .commit();
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    public static class SettingsFragment extends PreferenceFragmentCompat implements Preference.OnPreferenceChangeListener {
        @Override
        public void onCreatePreferencesFix(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey);
            EditTextPreference passwordPreference = findPreference("password");
            passwordPreference.setOnPreferenceChangeListener(this);
            ListPreference themePreference = findPreference("theme");
            themePreference.setOnPreferenceChangeListener(this);
        }

        @Override
        public boolean onPreferenceChange(Preference preference, Object newValue) {
            switch (preference.getKey()) {
                case "password":
                    EditText edit = ((EditTextPreference) preference).getEditText();
                    String pref = edit.getTransformationMethod().getTransformation(newValue.toString(), edit).toString();
                    preference.setSummary(pref);
                    break;
                case "theme":
                    AppCompatDelegate.setDefaultNightMode(Integer.parseInt((String) newValue));
                    break;
                default:
                    return false;
            }
            return true;
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        finish();
        return true;
    }
}

File: app/src/main/java/dev/voleum/ordermolder/ui/SettingsActivity.kt
package dev.voleum.ordermolder.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.appcompat.widget.Toolbar
import androidx.preference.ListPreference
import androidx.preference.Preference
import com.takisoft.preferencex.EditTextPreference
import com.takisoft.preferencex.PreferenceFragmentCompat
import dev.voleum.ordermolder.R
import dev.voleum.ordermolder.SettingsActivityJava


class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings, SettingsActivityJava.SettingsFragment())
                .commit()
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    class SettingsFragment : PreferenceFragmentCompat(), Preference.OnPreferenceChangeListener {

        override fun onCreatePreferencesFix(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey)
            val passwordPreference = findPreference<EditTextPreference>("password")
            setPasswordSummary(passwordPreference!!, passwordPreference.text)
            passwordPreference.onPreferenceChangeListener = this
            val themePreference = findPreference<ListPreference>("theme")
            themePreference!!.onPreferenceChangeListener = this
        }

        override fun onPreferenceChange(preference: Preference?, newValue: Any?): Boolean {
            when (preference!!.key) {
                "password" -> setPasswordSummary(preference as EditTextPreference, newValue as String)
                "theme"    -> AppCompatDelegate.setDefaultNightMode((newValue as String).toInt())
                else       -> return false
            }
            return true
        }

        private fun setPasswordSummary(preference: EditTextPreference, value: String?) {
            preference.editText.transformationMethod.getTransformation(value, preference.editText)
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}